test/EntityFramework6.Npgsql.Tests/PatternMatchingTests.cs

```diff
@@ -0,0 +1,147 @@
+﻿#region License
+// The PostgreSQL License
+//
+// Copyright (C) 2016 The Npgsql Development Team
+//
+// Permission to use, copy, modify, and distribute this software and its
+// documentation for any purpose, without fee, and without a written
+// agreement is hereby granted, provided that the above copyright notice
+// and this paragraph and the following two paragraphs appear in all copies.
+//
+// IN NO EVENT SHALL THE NPGSQL DEVELOPMENT TEAM BE LIABLE TO ANY PARTY
+// FOR DIRECT, INDIRECT, SPECIAL, INCIDENTAL, OR CONSEQUENTIAL DAMAGES,
+// INCLUDING LOST PROFITS, ARISING OUT OF THE USE OF THIS SOFTWARE AND ITS
+// DOCUMENTATION, EVEN IF THE NPGSQL DEVELOPMENT TEAM HAS BEEN ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+// THE NPGSQL DEVELOPMENT TEAM SPECIFICALLY DISCLAIMS ANY WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+// AND FITNESS FOR A PARTICULAR PURPOSE. THE SOFTWARE PROVIDED HEREUNDER IS
+// ON AN "AS IS" BASIS, AND THE NPGSQL DEVELOPMENT TEAM HAS NO OBLIGATIONS
+// TO PROVIDE MAINTENANCE, SUPPORT, UPDATES, ENHANCEMENTS, OR MODIFICATIONS.
+#endregion
+
+using Npgsql;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Data.Entity;
+using System.Linq;
+using System.Text;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Data.Entity.Core.Metadata.Edm;
+using System.Data.Entity.Core.Objects;
+using System.Data.Entity.Infrastructure;
+using NpgsqlTypes;
+using System.Text.RegularExpressions;
+
+namespace EntityFramework6.Npgsql.Tests
+{
+    class PatternMatchingTests : EntityFrameworkTestBase
+    {
+        // Comparisons are case-sensitive
+        [TestCase("blog", "blog", "BLOG")]
+
+        // The "^" and "$" language elements match the beginning and end of the input string
+        [TestCase("^blog$", "blog", "some \nblog\n name")]
+
+        // The "." language element matches every character except "\n"
+        [TestCase("some .* name", "some blog name", "some \n name")]
+
+        // Any white space in a regular expression pattern is interpreted as a literal space character
+        [TestCase("some blog name", "some blog name", "someblogname")]
```

Comment:
The empty spaces between the attributes are not needed, also the comments don't add value. It is better to use the ```TestName``` property of the TestCase attribute and set a really short descriptive name (eg: ```Case-sensitive``` for the first, ```^ and $ match beginning and end```, ```. matches all except \\n```, ```Whitespace not ignored in pattern```) so that it shows up in the test runner instead of the test parameters.

I also think the [Test] attribute should be added on the top of all the [TestCase] attributes (some test runners might need it).